Accessibility - Settings - Dismiss button

dusi opened the pull request:
# 📲 What

Adds accessibility label to the dismiss button & fixes the minimum width of the button in order to follow HIG best practices for tappable elements (44x44 pts)

# 🤔 Why

Better accessibility

# 🛠 How

Simply adding accessibility label (this will prevent from using the image name)

# 👀 See

| Before | After |
| --- | --- |
| <img width="782" alt="screen shot 2018-12-19 at 2 32 17 pm" src="https://user-images.githubusercontent.com/387596/50252468-f236c280-039b-11e9-946c-7f6c6ffcfc76.png"> | <img width="782" alt="screen shot 2018-12-19 at 2 33 59 pm" src="https://user-images.githubusercontent.com/387596/50252475-f7940d00-039b-11e9-868d-2575b45dcb06.png"> |
| <img width="785" alt="screen shot 2018-12-19 at 2 32 26 pm" src="https://user-images.githubusercontent.com/387596/50252482-ffec4800-039b-11e9-9cae-f97d3dd43609.png"> | <img width="781" alt="screen shot 2018-12-19 at 2 34 58 pm" src="https://user-images.githubusercontent.com/387596/50252485-05499280-039c-11e9-89f6-76a572380b1f.png"> |

# ✅ Acceptance criteria

- [x] Voice Over reads the dismiss button as "Dismiss - Button" (please not that this is localized)
- [x] Accessibility Inspector doesn't show min-width warning when running the audit on Settings screen

ifbarrera commented on Kickstarter-iOS/Views/Controllers/SettingsViewController.swift:
```diff
@@ -89,6 +84,19 @@ final class SettingsViewController: UIViewController {
+      target: self,
+      action: #selector(closeButtonPressed)
+    )
+    leftBarButtonItem.accessibilityLabel = Strings.Dismiss()
```
Is it possible to use the key path setters here?
`_ = leftBarButtonItem |> \.accessibilityLabel %~ { _ in Strings.Dismiss }`

dusi commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Audit Settings Accessibility](https://trello.com/c/xkODdosY/1125-audit-settings-accessibility)

dusi commented:
Addressed your feedback.

@ifbarrera could you please try the ACs on both iOS 10 / iOS 11 devices? Just to be sure nothing is out of ordinary. Our device list in YVR is very sparse at the moment.

ifbarrera commented:
Tested on iPhone 7 (10.3) and iPhone 8 (11.4) simulators using the accessibility inspector.
